src/wakefs/fuse.cpp

```diff
@@ -148,6 +149,49 @@ static bool collect_result_metadata(const std::string daemon_output, const struc
   return !result_ss.fail();
 }
 
+// Pre-resolve autofs paths through stating the paths on host machine
+static bool resolve_autofs_mounts(const std::vector<mount_op> &mount_ops) {
+  std::vector<std::string> autofs_paths;
+  for (const auto &op : mount_ops) {
+    if (op.type == "autofs-resolve") {
+      autofs_paths.push_back(op.source);
+    }
+  }
+
+  // Sort by path depth as we want parent paths resolved first for nested autofs situations
+  std::sort(autofs_paths.begin(), autofs_paths.end(),
+            [](const std::string &a, const std::string &b) {
+              auto count_directories = [](const std::string &path) {
+                if (path.empty() || path == "/") return 0;
+
+                std::vector<std::string> directories;
+                std::stringstream ss(path);
+                std::string directory;
+
+                while (std::getline(ss, directory, '/')) {
+                  if (!directory.empty()) {  // Skip slashes
+                    directories.push_back(directory);
+                  }
+                }
```

Comment:
I'm also a bit concerned by the potential for bad structures here.  Since it's all specifically under the autofs I guess we'd have to resolve it, and just rely on people building systems where Wake will be run being able to weigh the impact of autofs (and avoid the potential pathological setups), but for one example how does this handle a `/usr/child -> /usr/long/path/from/parent/autofs/to/child/autofs` symlink where both `/usr/.../parent` and `/usr/child` are listed in the `autofs_paths`?  (And even if symlinks are `realpath`ed, what about some similarly-degenerate bindmount?)